runtime/parachains/src/session_info.rs

```diff
@@ -120,7 +123,7 @@ impl<T: Config> Pallet<T> {
 		// create a new entry in `Sessions` with information about the current session
 		let new_session_info = SessionInfo {
 			validators, // these are from the notification and are thus already correct.
-			discovery_keys: take_active_subset(&active_set, &discovery_keys),
+			discovery_keys: take_active_subset_and_inactive(&active_set, &discovery_keys),
```

Comment:
This only seems to alter the discovery keys

Reply:
Yes, that's on purpose. We do have more authorities than parachain validators, but only parachain validators will be approval checkers, thus limiting assignment keys seems to be desired.